upa/src/sdk/sp1.ts

```diff
@@ -0,0 +1,244 @@
+  ),
+  GAMMA_NEG_Y_0: BigInt(
+    "11320632725496471124352232673686498720611893913179832806771034178631284892116"
+  ),
+  GAMMA_NEG_Y_1: BigInt(
+    "311895667756833756493099755182157054434816599621289498984920445818614464745"
+  ),
+
+  // Groth16 delta point in G2 in powers of i
+  DELTA_NEG_X_0: BigInt(
+    "1401586467118744686649898232509431431936958634904286407525795344840171509724"
+  ),
+  DELTA_NEG_X_1: BigInt(
+    "20524563347740346853186643374142185321405370469810755613088256350493061219989"
+  ),
+  DELTA_NEG_Y_0: BigInt(
+    "14665335796225261740324266854622393607627804871217389582109229884209518049091"
+  ),
+  DELTA_NEG_Y_1: BigInt(
+    "4675922494502640538519101370992755143525716960516481658013166874162416756018"
+  ),
+
+  // Constant and public input points
+  CONSTANT_X: BigInt(
+    "8310411565601441527035131496852221148476487246063333349863921483414575409313"
+  ),
+  CONSTANT_Y: BigInt(
+    "21343132819492496456805562863075138986248235983567056732949060829719030087739"
+  ),
+  PUB_0_X: BigInt(
+    "9193066544127521442507379487110183725565203847410097866655781919341371314500"
+  ),
+  PUB_0_Y: BigInt(
+    "1229953730407424098511641946145120096161248105904311204671469367759481075237"
+  ),
+  PUB_1_X: BigInt(
+    "9302931036688912050769082570627051329799904248848128887752496390619992168298"
+  ),
+  PUB_1_Y: BigInt(
+    "11368205644090635269179087945262736859378503013874730222437645132828283002424"
+  ),
+};
+/* eslint-enable max-len */
+
+export const sp1Versions = ["v1.2.0"];
+
+export const sp1VersionToGroth16Vk = new Map<string, SP1Groth16VerifyingKey>();
+
+sp1VersionToGroth16Vk.set("v1.2.0", SP1_GROTH16_VK_V1_2_0);
+
+export const convertSp1ProofFixture = (
+  fixture: SP1ProofFixture,
+  version: string
+) => {
+  if (!sp1Versions.includes(version)) {
+    throw new Error(`SP1 version ${version} not supported by UPA SDK`);
+  }
+
+  const groth16Vk = sp1VersionToGroth16Vk.get(version);
+
+  if (!groth16Vk) {
+    throw new Error(`SP1 version ${version} not supported by UPA SDK`);
+  }
+
+  const {
+    ALPHA_X,
+    ALPHA_Y,
+    BETA_NEG_X_0,
+    BETA_NEG_X_1,
+    BETA_NEG_Y_0,
+    BETA_NEG_Y_1,
+    GAMMA_NEG_X_0,
+    GAMMA_NEG_X_1,
+    GAMMA_NEG_Y_0,
+    GAMMA_NEG_Y_1,
+    DELTA_NEG_X_0,
+    DELTA_NEG_X_1,
+    DELTA_NEG_Y_0,
+    DELTA_NEG_Y_1,
+    CONSTANT_X,
+    CONSTANT_Y,
+    PUB_0_X,
+    PUB_0_Y,
+    PUB_1_X,
+    PUB_1_Y,
+  } = groth16Vk;
+
+  // Parse public inputs
+  const PROGRAM_VKEY = BigInt(fixture.vkey);
+  const PI_DIGEST = BigInt(fixture.publicValuesDigest);
+
+  const UPA_INPUTS = [PROGRAM_VKEY, PI_DIGEST];
+
+  // Parse proof
+  // Remove 0x prefix
+  const PROOF = fixture.proof.slice(2);
+
+  let idx = 0;
+
+  // First 4 bytes form the verifier hash
+  // See SP1Groth16Verifier.sol for details
+  // Currently unused in UPA SDK
+
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  const V_HASH = "0x" + PROOF.slice(idx, (idx += 8));
```

Comment:
I could be wrong, but I think we can just name unused vars with a _ at the front.

Reply:
Doesn't seem to work on TS. Definitely true on Rust.